src/Features/CSharp/Portable/CodeRefactorings/InlineTemporary/CSharpInlineTemporaryCodeRefactoringProvider.cs

```diff
@@ -88,36 +84,31 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
             if (!references.Any())
                 return;
 
+            // If the variable is itself referenced in its own initializer then don't offer anything here.  This
+            // practically does not occur (though the language allows it), and it only serves to add a huge amount
+            // of complexity to this feature.
+            if (references.Any(r => variableDeclarator.Initializer.Span.Contains(r.Span)))
+                return;
+
             context.RegisterRefactoring(
-                new MyCodeAction(
-                    c => InlineTemporaryAsync(document, variableDeclarator, c)),
+                new MyCodeAction(c => InlineTemporaryAsync(document, variableDeclarator, c)),
                 variableDeclarator.Span);
         }
 
         private static bool HasConflict(IdentifierNameSyntax identifier, VariableDeclaratorSyntax variableDeclarator)
         {
             // TODO: Check for more conflict types.
             if (identifier.SpanStart < variableDeclarator.SpanStart)
-            {
                 return true;
-            }
-
-            var identifierNode = identifier
-                .Ancestors()
-                .TakeWhile(n => n.Kind() is SyntaxKind.ParenthesizedExpression or SyntaxKind.CastExpression)
-                .LastOrDefault();
 
-            if (identifierNode == null)
-            {
-                identifierNode = identifier;
-            }
+            var identifierNode = identifier.WalkUpParentheses();
+            if (IsInDeconstructionAssignmentLeft(identifierNode))
+                return true;
```

Comment:
we had conflict detection happening at different times (which seems to have happened when deconstruction was added to the language).  This unifies things so that conflicts are all found in teh same place at teh same time and handled in the same way.